cmd/test-server/main.go

```diff
@@ -178,48 +196,164 @@ func pingHandler(w http.ResponseWriter, r *http.Request) {
 		panic(err)
 	}
 	stat := pinger.Statistics()
-	var icmp []RtItem
-	icmp = append(icmp, RtItem{clientIP, stat.PacketsSent, stat.PacketsRecv, stat.PacketLoss, fmtTimeMs(stat.MinRtt), fmtTimeMs(stat.AvgRtt), fmtTimeMs(stat.MaxRtt), fmtTimeMs(stat.StdDevRtt)})
+	icmp := RtItem{ip, stat.PacketsSent, stat.PacketsRecv, stat.PacketLoss, fmtTimeMs(stat.MinRtt), fmtTimeMs(stat.AvgRtt), fmtTimeMs(stat.MaxRtt), fmtTimeMs(stat.StdDevRtt)}
+	return icmp
+}
 
+func TcpPinger(ip string) tcpStruct {
 	// TCP Pinger
 	var tcpResultArr []tcpResult
 	rand.Seed(time.Now().UnixNano()) // Or each time we restart server the sequences would repeat
+	var portsWaitGroup sync.WaitGroup
+	portsWaitGroup.Add(len(PortsToTest))
 	for _, port := range PortsToTest {
-		var seqNumber uint64 = uint64(rand.Uint32())
-		var dst = fmt.Sprintf("%s:%d", clientIP, port)
-		ticker := time.NewTicker(TCPInterval)
-		var tResult []float64
-		for x := 0; x < TCPCounter; x++ {
-			seqNumber++
-			select {
-			case <-ticker.C:
-				tResult = append(tResult, pingTcp(dst, seqNumber, TCPTimeout))
+		go func(port int) {
+			defer portsWaitGroup.Done()
+			var seqNumber uint64 = uint64(rand.Uint32())
+			var dst = fmt.Sprintf("%s:%d", ip, port)
+			ticker := time.NewTicker(TCPInterval)
+			var tResult []float64
+			for x := 0; x < TCPCounter; x++ {
+				seqNumber++
+				select {
+				case <-ticker.C:
+					tResult = append(tResult, sendTcpPing(dst, seqNumber, TCPTimeout))
+				}
+			}
+			ticker.Stop()
+			min, avg, max, stddev := getStats(tResult)
+			tcpResultArr = append(tcpResultArr, tcpResult{port, tResult, min, avg, max, stddev})
+		}(port)
+	}
+	portsWaitGroup.Wait()
+	tcpResultObj := tcpStruct{ip, tcpResultArr}
+	return tcpResultObj
+}
+
+// Avg RTT from all successful ICMP measurements, to display on webpage
+func getAvgIcmpResult(icmp []RtItem) float64 {
```

Comment:
Suggestion for a more specific name:

```suggestion
func getMeanIcmpRTT(icmp []RtItem) float64 {
```